### ISSUE TITLE
Deprecated conversion introduced in Rails 7.0 is incompatible with implicit default format

### ISSUE THREAD

**Issue by wata727** [rails/rails#48545]
### Steps to reproduce

We are planning to upgrade an application running on Rails 6.1 to Rails 7.0. Among them, we are looking at the deprecated conversions introduced in #43772.

I'm guessing this was probably designed to be backwards compatible as of Rails 7.0, but if we've added `Date::DATE_FORMATS[:default]`, the result changed in Rails 7.0.

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "activesupport", "7.0.5"
end

require "active_support"
require "active_support/core_ext/date"
require "minitest/autorun"

# Add default format
Date::DATE_FORMATS[:default] = '%Y/%m/%d'

class DateDeprecatedConversionsTest < Minitest::Test
  def test_implicit_default_conversions
    assert_equal Date.new(2023, 6, 21).to_s, '2023/06/21'
  end

  def test_explicit_default_conversions
    assert_equal Date.new(2023, 6, 21).to_s(:default), '2023/06/21'
  end
end
```

### Expected behavior

These tests are passed in Rails 6.1. So it should work on Rails 7.0 as well.

```diff
diff --git a/test.rb b/test.rb
index 187c8500dd6..9cb2e93c834 100644
--- a/test.rb
+++ b/test.rb
@@ -8,7 +8,7 @@ gemfile(true) do
   git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
   # Activate the gem you are reporting the issue against.
-  gem "activesupport", "7.0.5"
+  gem "activesupport", "6.1.7.3"
 end
 
 require "active_support"
```

```
# Running:

..

Finished in 0.001116s, 1792.1149 runs/s, 1792.1149 assertions/s.

2 runs, 2 assertions, 0 failures, 0 errors, 0 skips
```

### Actual behavior

Implicit default format does not work in Rails 7.0.

```
# Running:

DEPRECATION WARNING: Date#to_s(:default) is deprecated. Please use Date#to_fs(:default) instead. (called from test_explicit_default_conversions at test.rb:27)
.F

Finished in 0.101925s, 19.6223 runs/s, 19.6223 assertions/s.

  1) Failure:
DateDeprecatedConversionsTest#test_implicit_default_conversions [test.rb:23]:
--- expected
+++ actual
@@ -1,3 +1 @@
-# encoding: US-ASCII
-#    valid: true
-"2023-06-21"
+"2023/06/21"


2 runs, 2 assertions, 1 failures, 0 errors, 0 skips
```

Perhaps calling `to_formatted_s(:default)` instead of `to_default_s` here would keep compatibility.

https://github.com/rails/rails/blob/e88857bbb9d4e1dd64555c34541301870de4a45b/activesupport/lib/active_support/core_ext/date/deprecated_conversions.rb#L18

However, [this conversion has been removed](https://github.com/rails/rails/commit/e420c3380eb2b698a4fe84ed196f914d18f7844a) in the main branch, and adding the `Date::DATE_FORMATS[:default]` may be undesirable behaviour (In general, I understand that this is a bad practice). I just wanted to share this behavior and discuss whether it's worth fixing it in the Rails 7.0 branch.

### System configuration
**Rails version**:
6.1.7.3/7.0.5

**Ruby version**:
3.1.4

**Comment by skipkayhil** [comment]
Hey @wata727, thanks for opening an issue! I've opened a PR to fix this: #48555. Can you test it out and let me know if it fixes the problem?

**Comment by wata727** [comment]
Thank you for working on this! It seems to work fine in our application.
